### PR TITLE
Error handling improvements

### PR DESCRIPTION
**Pull request by krzysztofziobro**
In this PR:
- automated restart of AzeroListener in case of pause + unpause
- different way of overriding redis cache that allows for previous
- added sleep to the advisory handling: previous version would needlessly spam node with requests
- removed special treatment of CommitteeId mismatch

### CODE REVIEW

File: relayer/relayer/src/main.rs [hunk 1]
```diff
@@ -139,47 +144,148 @@ async fn main() -> Result<()> {
...
+        // For most errors we don't want to automatically restart the relayer but wait for manual intervention
+        loop {
+            error!("Error when running listeners, this might require manual investigation or RESTART...");
+            err.chain().enumerate().for_each(|(level, cause)| {
+                let cause = cause.to_string();
+                if cause.len() > 100 {
+                    error!(" {}: {}...", level, &cause[..100]);
+                } else {
+                    error!(" {}: {}", level, cause);
+                }
```
Comment (obrok) [inline relayer/relayer/src/main.rs:166]: Suggestion: print the whole thing. Why are you worried about `cause` being longer than 100?

**Review by obrok — APPROVED** [review]
One suggestion, but may be irrelevant if you just Let It Crash™ as discussed.

File: relayer/relayer/src/main.rs (hunk 1 above)
Comment (obrok) [reply relayer/relayer/src/main.rs]: Hm, unfortunate, but maybe we can only shorten that particular one? Better yet - if we wrap it in some error of our own, implement the shortening in there.

Comment (krzysztofziobro) [reply relayer/relayer/src/main.rs]: Changed it to print the whole error once

**Pull request merged**